File: 10-04-2024/Party of Couples.py
#User function Template for python3

class Solution:
    def findSingle(self, n, arr):
        #Your Code Start
        var = 0
        for i in range(n):
            var = var ^ arr[i]
        return var
        # Your Code End
      


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        N = int(input())
        arr = input().split()
        for itr in range(N):
            arr[itr] = int(arr[itr])
        
        ob = Solution()
        print(ob.findSingle(N, arr))

# } Driver Code Ends

File: 10-04-2024/Party of Couples.cs
//{ Driver Code Starts
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                int N = Convert.ToInt32(Console.ReadLine());
                int[] arr = new int[N];
                var stringArray = Console.ReadLine().Split(' ');
                for (int i = 0; i < N; i++)
                {
                    arr[i] = Convert.ToInt32(stringArray[i]);
                }

                Solution obj = new Solution();
                int res = obj.findSingle(N, arr);
                Console.Write(res);
                Console.Write("\n");
          }

        }
    }
}
// } Driver Code Ends


//User function Template for C#

class Solution
    {
        //Complete this function
        public int findSingle(int N, int[] arr)
        {
            //Your Code Start
           int var = 0;
           for (int i = 0; i < N; i++) {
              var ^= arr[i];
             }
           return var;
           

            //Your code End
        }
    }
